app/src/main/java/com/github/capntrips/kernelflasher/ui/screens/slot/SlotViewModel.kt

```diff
@@ -433,7 +434,9 @@ class SlotViewModel(
     private fun resetSlot() {
         val activeSlotSuffix = Shell.cmd("getprop ro.boot.slot_suffix").exec().out[0]
         val newSlot = if (activeSlotSuffix == "_a") "_b" else "_a"
-        Shell.cmd("magisk resetprop -n ro.boot.slot_suffix $newSlot").exec()
+        val isMagisk = Shell.cmd("which magisk").exec().isSuccess;
+        val resetprop = if (isMagisk) "magisk resetprop" else "resetprop"
+        Shell.cmd("$resetprop -n ro.boot.slot_suffix $newSlot").exec()
```

Comment:
I realize this method already existed within the codebase, but I only recently learned that the AK3 slot can be overridden. The `999-flashafterupdate.sh` script [here](https://forum.xda-developers.com/t/tools-zips-scripts-osm0sis-odds-and-ends-multiple-devices-platforms.2239421/post-40409361) sets AK3 up for flashing to the inactive slot during an OTA. I haven't tested it, but the important bit seems to be:

```
case $slot in
  _a) slot=_b;;
  _b) slot=_a;;
esac
```
I imagine something like this would be cleaner than `resetprop`.

Again, I can look into this in the next few days if this isn't something you immediately want to work on.